test/jasmine/tests/gl3d_plot_interact_test.js

```diff
@@ -66,7 +66,7 @@ describe('Test gl3d plots', function() {
         gd = createGraphDiv();
         ptData = {};
 
-        jasmine.DEFAULT_TIMEOUT_INTERVAL = 4000;
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = 8000;
```

Comment:
Hmm. I was about to ask if you noticed any performance deterioration on your branch. I guess this here answers my question. :thinking:

Reply:
... would there be a way to increase the resolution just for traces that _need it_? In other words, is a way to determine if a given `z` 2D array requires extra resolution?

Reply:
One idea could be to allow resetting this resolution via the api.
I can imagine of auto-detect solutions; but considering animations I suggest we keep this constant.

Reply:
@etpinard Rather than using `1024`, now by simply using `120` instead of `128' the performance is untouched; while the quality of the graphs are remarkably improved. If we want to go further than that the next magic values that could be applied are 180, 240, 360, 720 and 840.

Reply:
Fantastic! Could you explain briefly why `120` is better than `128`?

Reply:
128 = 2<sup>7</sup> so it is divisible by (1 + 7) numbers {1, 2, 4, 8, 16, 32, 64, 128}
120 = 2<sup>3</sup>x3<sup>1</sup>x5<sup>1</sup> therefore divisible by (1+3)x(1+1)x(1+1) = 16 numbers {1, 2, 3, 4, 5, 6, 8, 10, 12, 15, 16, 20, 24, 30, 60, 120}.
So it is more likely that the refined grid match actual data points in this case.
https://en.wikipedia.org/wiki/Highly_composite_number
Base on this investigation I am now thinking to rewriting the function that refines scale.

Reply:
Wow @archmoj is the man! Thanks!